plugins/modules/idrac_session.py

```diff
@@ -0,0 +1,429 @@
+    """
+    def execute(self):
+        """
+        Executes the session creation process.
+
+        This function creates a session by sending a POST request to the session URL with the
+        provided username and password.
+        If the request is successful (status code 201), it retrieves the session details, removes
+        any OData keys from the response,
+        and extracts the X-Auth-Token from the response headers. It then exits the module with a
+        success message, indicating that
+        the session was created successfully, and provides the session data and X-Auth-Token as
+        output variables.
+
+        If the request fails (status code other than 201), it exits the module with a failure
+        message, indicating that the session creation failed.
+
+        Parameters:
+            None
+
+        Returns:
+            None
+        """
+        payload = {}
+        payload = {"UserName": self.module.params.get("username"),
+                   "Password": self.module.params.get("password")}
+        session_url = self.get_session_url()
+        if self.module.check_mode:
+            self.module.exit_json(msg=CHANGES_FOUND_MSG, changed=True)
+        session_response = self.idrac.invoke_request(session_url, "POST", data=payload)
+        status = session_response.status_code
+        if status == 201:
+            session_details = session_response.json_data
+            session_data = remove_key(session_details, regex_pattern=ODATA_REGEX)
+            x_auth_token = session_response.headers.get('X-Auth-Token')
+            self.module.exit_json(msg=CREATE_SUCCESS_MSG,
+                                  changed=True,
+                                  session_data=session_data,
+                                  x_auth_token=x_auth_token)
+        else:
+            self.module.exit_json(msg=CREATE_FAILURE_MSG, failed=True)
+
+
+class DeleteSession(Session):
+    """
+    Deletes a session.
+    """
+    def execute(self):
+        """
+        Executes the deletion of a session.
+
+        This function retrieves the session ID from the module parameters and constructs the
+        session URL using the `get_session_url` method. It then invokes a DELETE request to the
+        session URL with the session ID appended. The response from the request is stored in the
+        `session_response` variable.
+
+        If the response status code is 200, indicating a successful deletion, the function exits
+        the module with a success message and sets the `changed` parameter to True. Otherwise, it
+        exits the module with a failure message and sets the `failed` parameter to True.
+
+        Parameters:
+            None
+
+        Returns:
+            None
+        """
+        session_id = self.module.params.get("session_id")
+        session_url = self.get_session_url()
+        if self.module.check_mode:
+            session_status = self.__get_session_status(session_url, session_id)
```

Comment:
We can take this out of if-else conditions as it is being used in both.

Reply:
Done